### ISSUE TITLE
petsc@3.21.2 +rocm w/ ROCm 6.1.1, error unclear?

### ISSUE THREAD

**Issue by eugeneswalker** [LLNL/spack#44600]
### Steps to reproduce the issue

`petsc@3.21.2 +rocm` build fails using:
* spack@develop (641075539c0a888d4378bd7dfdb19a872c5c42de from `Mon Jun 3 14:37:51 2024 -0700`)
* ROCm 6.1.1 installed via apt, externally defined via `packages:` config
* Ubuntu 22.04, X86_64

It's not clear to me what the failure is about...

Spack environment: [spack.yaml.txt](https://github.com/user-attachments/files/15687467/spack.yaml.txt)

<details><summary>Concretization</summary>

<pre>
 -   ncqyiqv  petsc@3.21.2%gcc@11.4.0~X~batch~cgns~complex~cuda~debug+double~exodusii~fftw+fortran~giflib+hdf5~hpddm~hwloc+hypre~int64~jpeg~knl~kokkos~libpng~libyaml~memkind+metis~mkl-pardiso~mmg~moab~mpfr+mpi~mumps~openmp~p4est~parmmg~ptscotch~random123+rocm~saws~scalapack+shared~strumpack~suite-sparse+superlu-dist~sycl~tetgen~trilinos~valgrind~zoltan amdgpu_target=gfx90a build_system=generic clanguage=C memalign=none arch=linux-ubuntu22.04-x86_64_v3
[+]  sycoqix      ^diffutils@3.10%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  hshzy76      ^gcc-runtime@11.4.0%gcc@11.4.0 build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[e]  a7drdl4      ^glibc@2.35%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  aevr47s      ^gmake@4.4.1%gcc@11.4.0~guile build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[+]  tbbf4mx      ^hdf5@1.14.3%gcc@11.4.0~cxx~fortran~hl~ipo~java~map+mpi+shared~subfiling~szip~threadsafe+tools api=default build_system=cmake build_type=Release generator=make patches=82088c8 arch=linux-ubuntu22.04-x86_64_v3
[+]  45jczu5          ^cmake@3.29.2%gcc@11.4.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-ubuntu22.04-x86_64_v3
[+]  7wjyjza              ^curl@8.7.1%gcc@11.4.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-ubuntu22.04-x86_64_v3
[+]  zc6oo3e                  ^nghttp2@1.62.0%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  mv2wke5          ^pkgconf@2.2.0%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[e]  abm43bu      ^hip@6.1.1%gcc@11.4.0~cuda+rocm build_system=cmake build_type=Release generator=make patches=1f65dfe,6bbe83b arch=linux-ubuntu22.04-x86_64_v3
[e]  h7o5v3r      ^hipblas@6.1.1%gcc@11.4.0~asan~cuda+rocm amdgpu_target=auto build_system=cmake build_type=Release generator=make patches=b05b34b arch=linux-ubuntu22.04-x86_64_v3
[e]  yof52ip      ^hipsolver@6.1.1%gcc@11.4.0~asan~cuda+rocm amdgpu_target=auto build_system=cmake build_type=Release generator=make patches=286e898 arch=linux-ubuntu22.04-x86_64_v3
[e]  vmg5y2y      ^hipsparse@6.1.1%gcc@11.4.0~asan~cuda+rocm amdgpu_target=auto build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64_v3
[e]  qyqwwse      ^hsa-rocr-dev@6.1.1%gcc@11.4.0~asan+image+shared build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64_v3
[+]  gnaixth      ^hypre@2.31.0%gcc@11.4.0~caliper~complex~cublas~cuda~debug+fortran~gptune~gpu-aware-mpi~int64~internal-superlu~magma~mixedint+mpi~openmp~rocblas~rocm+shared~superlu-dist~sycl~umpire~unified-memory build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[e]  l4maeaq      ^llvm-amdgpu@6.1.1%gcc@11.4.0~link_llvm_dylib~llvm_dylib+rocm-device-libs build_system=cmake build_type=Release generator=ninja patches=b4774ca arch=linux-ubuntu22.04-x86_64_v3
[+]  mz6trb5      ^metis@5.1.0%gcc@11.4.0~gdb~int64~ipo~real64+shared build_system=cmake build_type=Release generator=make patches=4991da9,93a7903,b1225da arch=linux-ubuntu22.04-x86_64_v3
[+]  q73zij2      ^mpich@4.2.1%gcc@11.4.0~argobots~cuda+fortran~hwloc+hydra+libxml2+pci~rocm+romio~slurm~vci~verbs~wrapperrpath~xpmem build_system=autotools datatype-engine=auto device=ch4 netmod=ofi pmi=default arch=linux-ubuntu22.04-x86_64_v3
[+]  43jtbdg          ^findutils@4.9.0%gcc@11.4.0 build_system=autotools patches=440b954 arch=linux-ubuntu22.04-x86_64_v3
[+]  grokknt          ^libfabric@1.21.0%gcc@11.4.0~debug~kdreg~uring build_system=autotools fabrics=sockets,tcp,udp arch=linux-ubuntu22.04-x86_64_v3
[+]  cinntmr          ^libpciaccess@0.17%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  ekglkxs              ^libtool@2.4.7%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  eu34nvi              ^util-macros@1.20.1%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  blvm6hb          ^libxml2@2.10.3%gcc@11.4.0+pic~python+shared build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  x2wkmjd          ^yaksa@0.3%gcc@11.4.0~cuda~rocm build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  tfy5aww              ^autoconf@2.72%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  c5h3rwv              ^automake@1.16.5%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  iloyaza              ^m4@1.4.19%gcc@11.4.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu22.04-x86_64_v3
[+]  jv4abt3                  ^libsigsegv@2.14%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  na3654n      ^openblas@0.3.26%gcc@11.4.0~bignuma~consistent_fpcsr+dynamic_dispatch+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=openmp arch=linux-ubuntu22.04-x86_64_v3
[+]  vqvzn64          ^perl@5.38.2%gcc@11.4.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-ubuntu22.04-x86_64_v3
[+]  sw7gp6x              ^berkeley-db@18.1.40%gcc@11.4.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu22.04-x86_64_v3
[+]  jsqcje5      ^parmetis@4.0.3%gcc@11.4.0~gdb~int64~ipo+shared build_system=cmake build_type=Release generator=make patches=4f89253,50ed208,704b84f arch=linux-ubuntu22.04-x86_64_v3
[+]  rmokzl2      ^python@3.11.9%gcc@11.4.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=13fa8bf,b0615b2,ebdca64,f2fd060 arch=linux-ubuntu22.04-x86_64_v3
[+]  tmq3cku          ^bzip2@1.0.8%gcc@11.4.0~debug~pic+shared build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[+]  6sx3gbp          ^expat@2.6.2%gcc@11.4.0+libbsd build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  wfvladn              ^libbsd@0.12.2%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  i3snxv7                  ^libmd@1.0.4%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  xucr42b          ^gdbm@1.23%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  3ea5rwn          ^gettext@0.22.5%gcc@11.4.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  a4yvhcg              ^tar@1.34%gcc@11.4.0 build_system=autotools zip=pigz arch=linux-ubuntu22.04-x86_64_v3
[+]  5qg7ajj                  ^pigz@2.8%gcc@11.4.0 build_system=makefile arch=linux-ubuntu22.04-x86_64_v3
[+]  fz3fbgw                  ^zstd@1.5.6%gcc@11.4.0+programs build_system=makefile compression=none libs=shared,static arch=linux-ubuntu22.04-x86_64_v3
[+]  ounz353          ^libffi@3.4.6%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  wq6feyo          ^libxcrypt@4.4.35%gcc@11.4.0~obsolete_api build_system=autotools patches=4885da3 arch=linux-ubuntu22.04-x86_64_v3
[+]  sqasgz3          ^ncurses@6.5%gcc@11.4.0~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=linux-ubuntu22.04-x86_64_v3
[+]  r3wk5bb          ^openssl@3.3.0%gcc@11.4.0~docs+shared build_system=generic certs=mozilla arch=linux-ubuntu22.04-x86_64_v3
[+]  t2brqss              ^ca-certificates-mozilla@2023-05-30%gcc@11.4.0 build_system=generic arch=linux-ubuntu22.04-x86_64_v3
[+]  bo6z4bc          ^readline@8.2%gcc@11.4.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu22.04-x86_64_v3
[+]  jvljkqx          ^sqlite@3.43.2%gcc@11.4.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  vl2hqwb          ^util-linux-uuid@2.40.1%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
[+]  imwcbq7          ^xz@5.4.6%gcc@11.4.0~pic build_system=autotools libs=shared,static arch=linux-ubuntu22.04-x86_64_v3
[e]  oha4l4u      ^rocblas@6.1.1%gcc@11.4.0+tensile amdgpu_target=auto build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64_v3
[e]  dblglzl      ^rocprim@6.1.1%gcc@11.4.0 amdgpu_target=auto build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64_v3
[e]  oqatonm      ^rocrand@6.1.1%gcc@11.4.0+hiprand amdgpu_target=auto build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64_v3
[e]  mav7pps      ^rocsolver@6.1.1%gcc@11.4.0+optimal amdgpu_target=auto build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64_v3
[e]  xy65h5f      ^rocsparse@6.1.1%gcc@11.4.0~test amdgpu_target=auto build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64_v3
[e]  dgow4hb      ^rocthrust@6.1.1%gcc@11.4.0 amdgpu_target=auto build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64_v3
[+]  xg6nhdx      ^superlu-dist@8.2.1%gcc@11.4.0~cuda~int64~ipo~openmp+parmetis~rocm+shared build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64_v3
[+]  54554gr      ^zlib-ng@2.1.6%gcc@11.4.0+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-ubuntu22.04-x86_64_v3
</pre>
</details>

Error:
```
...
==> Installing petsc-3.21.2-ncqyiqvri5cjc2v3y4fy6slpvdzdniya [51/51]
==> No binary for petsc-3.21.2-ncqyiqvri5cjc2v3y4fy6slpvdzdniya found: installing from source
==> Fetching http://web.cels.anl.gov/projects/petsc/download/release-snapshots/petsc-3.21.2.tar.gz
==> No patches needed for petsc
==> petsc: Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 1:
    '/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/python-3.11.9-rmokzl2xfjfmhw67nzulwle6hmcd5p2p/bin/python3.11' 'configure' '--prefix=/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/petsc-3.21.2-ncqyiqvri5cjc2v3y4fy6slpvdzdniya' '--with-ssl=0' '--download-c2html=0' '--download-sowing=0' '--download-hwloc=0' '--with-make-exec=make' '--with-cc=/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/mpich-4.2.1-q73zij2acabvp27dqu6eakamcxfxcsja/bin/mpicc' '--with-cxx=/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/mpich-4.2.1-q73zij2acabvp27dqu6eakamcxfxcsja/bin/mpic++' '--with-fc=/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/mpich-4.2.1-q73zij2acabvp27dqu6eakamcxfxcsja/bin/mpif90' '--with-precision=double' '--with-scalar-type=real' '--with-shared-libraries=1' '--with-debugging=0' '--with-openmp=0' '--with-64-bit-indices=0' '--with-blas-lapack-lib=/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/openblas-0.3.26-na3654nrvh23utovqzemdre3o7bnrutg/lib/libopenblas.so' '--with-x=0' '--with-sycl=0' '--with-clanguage=C' '--with-cuda=0' '--with-hip=1' '--with-hip-include=/opt/rocm-6.1.1/include' '--with-metis=1' '--with-metis-include=/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/metis-5.1.0-mz6trb5ztvggljs566dlu5n7cvnz7u5r/include' '--with-metis-lib=/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/metis-5.1.0-mz6trb5ztvggljs566dlu5n7cvnz7u5r/lib/libmetis.so' '--with-hypre=1' '--with-hypre-include=/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/hypre-2.31.0-gnaixthhgc3vjmio4cxlg47oskqr5ifo/include' '--with-hypre-lib=/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/hypre-2.31.0-gnaixthhgc3vjmio4cxlg47oskqr5ifo/lib/libHYPRE.so' '--with-parmetis=1' '--with-parmetis-include=/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/parmetis-4.0.3-jsqcje5ptx7lslpfz7ws7pijylxybx6k/include' '--with-parmetis-lib=/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/parmetis-4.0.3-jsqcje5ptx7lslpfz7ws7pijylxybx6k/lib/libparmetis.so' '--with-kokkos=0' '--with-kokkos-kernels=0' '--with-superlu_dist=1' '--with-superlu_dist-include=/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/superlu-dist-8.2.1-xg6nhdxsspwl6hzfuza7aupcibskrt4r/include' '--with-superlu_dist-lib=/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/superlu-dist-8.2.1-xg6nhdxsspwl6hzfuza7aupcibskrt4r/lib/libsuperlu_dist.so' '--with-ptscotch=0' '--with-suitesparse=0' '--with-hdf5=1' '--with-hdf5-include=/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/hdf5-1.14.3-tbbf4mxnxfazfahtoar2k5hmcdyhrzh6/include' '--with-hdf5-lib=/spack/opt/spack/linux-ubuntu22.04-x86_64_v3/gcc-11.4.0/hdf5-1.14.3-tbbf4mxnxfazfahtoar2k5hmcdyhrzh6/lib/libhdf5.so' '--with-zlib=0' '--with-mumps=0' '--with-trilinos=0' '--with-fftw=0' '--with-valgrind=0' '--with-gmp=0' '--with-libpng=0' '--with-giflib=0' '--with-mpfr=0' '--with-netcdf=0' '--with-pnetcdf=0' '--with-moab=0' '--with-random123=0' '--with-exodusii=0' '--with-cgns=0' '--with-memkind=0' '--with-p4est=0' '--with-saws=0' '--with-yaml=0' '--with-hwloc=0' '--with-libjpeg=0' '--with-scalapack=0' '--with-strumpack=0' '--with-mmg=0' '--with-parmmg=0' '--with-tetgen=0' '--with-zoltan=0' '--with-hip-arch=gfx90a' 'HIPPPFLAGS=-I/opt/rocm-6.1.1/include -I/opt/rocm-6.1.1/include -I/opt/rocm-6.1.1/include -I/opt/rocm-6.1.1/include -I/opt/rocm-6.1.1/include -I/opt/rocm-6.1.1/include -I/opt/rocm-6.1.1/include -I/opt/rocm-6.1.1/include -I/opt/rocm-6.1.1/include ' '--with-hip-lib=/opt/rocm-6.1.1/lib/libhipsparse.so /opt/rocm-6.1.1/lib/libhipblas.so /opt/rocm-6.1.1/lib/libhipsolver.so /opt/rocm-6.1.1/lib/librocsparse.so /opt/rocm-6.1.1/lib/librocsolver.so /opt/rocm-6.1.1/lib/librocblas.so  -L/opt/rocm-6.1.1/lib -lamdhip64'
See build log for details:
  /tmp/root/spack-stage/spack-stage-petsc-3.21.2-ncqyiqvri5cjc2v3y4fy6slpvdzdniya/spack-build-out.txt
```

### Error message

<details><summary>Error message</summary>

<pre>
... see above
</pre></details>

### Information on your system

* **Spack:** 0.23.0.dev0 (136308c8fdc3d3d0593d563103f34e60ab15a5d6)
* **Python:** 3.10.12
* **Platform:** linux-ubuntu22.04-cascadelake
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/user-attachments/files/15687425/spack-build-env.txt)
[spack-build-out.txt](https://github.com/user-attachments/files/15687428/spack-build-out.txt)
[configure.log](https://github.com/user-attachments/files/15687491/configure.log)

@jedbrown @BarrySmith @balay 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by balay** [comment]
Looks like rocm-6.1 is released after  petsc-3.21 - its fix is in petsc/main via https://gitlab.com/petsc/petsc/-/merge_requests/7523

cc: @jczhang07

**Comment by eugeneswalker** [comment]
This is fixed by https://github.com/spack/spack/pull/46486
